safety_module/src/lib.rs

```diff
@@ -254,16 +253,21 @@ pub trait SafetyModule {
             .esdt_local_burn(&token_identifier, nonce, &amount);
     }
 
-    fn mint_deposit_nft(self, deposit_metadata: &DepositMetadata, amount: Self::BigUint) {
-        self.send().esdt_nft_create::<DepositMetadata>(
-            &self.nft_token().get(),
-            &amount,
-            &BoxedBytes::empty(),
-            &Self::BigUint::zero(),
-            &BoxedBytes::empty(),
-            deposit_metadata,
-            &[BoxedBytes::empty()],
-        );
+    fn mint_deposit_nft(
+        self,
+        deposit_metadata: &DepositPosition<Self::BigUint>,
+        amount: Self::BigUint,
+    ) {
+        self.send()
+            .esdt_nft_create::<DepositPosition<Self::BigUint>>(
```

Comment:
No need to have the type here, the Rust compiler can deduce it.
```suggestion
            .esdt_nft_create(
```

Reply:
done